packages/attestation-service/src/request.ts

```diff
@@ -0,0 +1,83 @@
+import { isE164NumberStrict } from '@celo/utils/lib/phoneNumbers'
+import { isValidAddress } from '@celo/utils/lib/signatureUtils'
+import express from 'express'
+import { either, isLeft } from 'fp-ts/lib/Either'
+import * as t from 'io-ts'
+
+export function parseRequest<T>(
```

Comment:
Is this some kind of fp-ts pattern? This structure of a function that returns a function which uses another function as a param seems more complex than it has to be.

Reply:
I actually tried to minimize the style impact of fp-ts here, so this should mostly be a single-level function composition. Basically, when one writes an express handler, a developer would have to do some amount of logic to validate that the request has the right payload, types, etc.

To avoid that duplicated logic, a developer could use `parseRequest(RequestType, processor)` to receive the parsed request as a third argument. So IMO you get the benefit of declarative routes in https://github.com/celo-org/celo-monorepo/pull/1468/files#diff-8b9f369026f23d51728f7ad563386fdcR22, while the actual request handler has to not deal with the validation of the request payload and can just rely on it been validated by `parseRequest` https://github.com/celo-org/celo-monorepo/pull/1468/files#diff-881b8cc344bdb8ee0e96c73fb379ee6fR94

Reply:
Okay seems reasonable. I guess I may just have been thrown off by the names them. Maybe `handler` is a more conventional name for `processor` and `createValidatedHandler` is a more accurate name than `parseRequest`? Unless these are fp-ts terms

Reply:
Nope, they are just @nambrot™ , so changed them to your suggestion!